src/fondant/compiler.py

```diff
@@ -305,3 +305,83 @@ def _set_configuration(self, task, fondant_component_operation):
             task.add_node_selector_constraint("node_pool", node_pool_name)
 
         return task
+
+
+class VertexCompiler(Compiler):
+    def __init__(self):
+        self.resolve_imports()
+
+    def resolve_imports(self):
+        """Resolve imports for the Vertex compiler."""
+        try:
+            import kfp
+
+            self.kfp = kfp
+
+        except ImportError:
+            msg = """You need to install kfp to use the Vertex compiler,\n
+                     you can install it with `pip install fondant[vertex]`"""
+            raise ImportError(
+                msg,
+            )
+
+    def compile(
+        self,
+        pipeline: Pipeline,
+        output_path: str = "vertex_pipeline.yml",
+    ) -> None:
+        """Compile a pipeline to vertex pipeline spec and save it to a specified output path.
+
+        Args:
+            pipeline: the pipeline to compile
+            output_path: the path where to save the Kubeflow pipeline spec
+        """
+        self.pipeline = pipeline
+        self.pipeline.validate(run_id="{{workflow.name}}")
+        logger.info(f"Compiling {self.pipeline.name} to {output_path}")
+
+        @self.kfp.dsl.pipeline(name=pipeline.name, description=pipeline.description)
+        def kfp_pipeline():
+            previous_component_task = None
+            manifest_path = None
+            for component_name, component in self.pipeline._graph.items():
+                logger.info(f"Compiling service for {component_name}")
+
+                component_op = component["fondant_component_op"]
+                # convert ComponentOp to Kubeflow component
+                kubeflow_component_op = self.kfp.components.load_component_from_text(
```

Comment:
It seems like this is the only difference between the two compiler (the SDK changed for this function). Can we perhaps reuse the same function instead of defining it twice. Seems like there is quite a bit of overlap